solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -219,6 +221,28 @@ public boolean isLazyFieldLoadingEnabled() {
     }
   }
 
+  /** Returns a collection of the names of all indexed fields which the index reader knows about. */
+  public Collection<String> getIndexedFieldNames() {
+    synchronized (this) {
+      if (indexedFieldNames == null) {
+        indexedFieldNames = new LinkedList<>();
+        for (FieldInfo fieldInfo : searcher.getFieldInfos()) {
+          final String fieldName = fieldInfo.name;
+          try {
+            SchemaField field = searcher.getSchema().getField(fieldName);
```

Comment:
This isn't necessary; can't we look at FieldInfo for this without consulting Solr's schema?

Reply:
Changed in https://github.com/apache/solr/pull/757/commits/58abfdd6557700585b8abfa9c8ef698e185d74f6 commit.